app/src/main/java/org/schabi/newpipe/util/StreamDialogEntry.java

```diff
@@ -206,4 +229,30 @@ private static void openChannelFragment(final Fragment fragment,
                 fragment.requireActivity().getSupportFragmentManager(),
                 item.getServiceId(), uploaderUrl, item.getUploaderName());
     }
+
+    private static boolean isNetworkAvailable(final Context context) {
+        final ConnectivityManager cm =
+                (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+
+        final NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
+        return activeNetwork != null
+                && activeNetwork.isConnectedOrConnecting();
+    }
```

Comment:
Note for a future PR: this method and all of the other methods related to `ConnectivityManager` distributed across the app should be moved into a separate `util.` class.